src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -514,6 +514,28 @@ void ClusterResourceScheduler::DeleteResource(const std::string &node_id_string,
   }
 }
 
+bool ClusterResourceScheduler::ContainResource(const std::string &resource_name) {
+  int idx = -1;
+  if (resource_name == ray::kCPU_ResourceLabel) {
+    idx = (int)CPU;
+  } else if (resource_name == ray::kGPU_ResourceLabel) {
+    idx = (int)GPU;
+  } else if (resource_name == ray::kObjectStoreMemory_ResourceLabel) {
+    idx = (int)OBJECT_STORE_MEM;
+  } else if (resource_name == ray::kMemory_ResourceLabel) {
+    idx = (int)MEM;
+  };
+  if (idx != -1) {
```

Comment:
Can we just check `local_resources_ .predefined_resources.find(...)`? (That seems to be more robust in the long term)

Reply:
The current implementation has some assumption that can be broken if we develop new features.

Reply:
I think it may be problematic if we use `local_resources_ .predefined_resources.find(...)` here directly after I checked the two structs `ResourceCapacity` and `ResourceInstanceCapacities`.
 For example with the `ResourceCapacity`, We can't make sure it's the same `ResourceCapacity` even if the total value and the available value are the same, right?

Reply:
Can we do this in a separate pr?  I actually don't want to involve more unrelated changes to the current pr...

Reply:
discussed offline.